CHANGELOG.md

```diff
@@ -3,13 +3,143 @@
+
+```diff
+- plugins:
+-   experimental.traffic_shaping:
+-     variables_deduplication: true # Enable the variables deduplication optimization
+-     all:
+-       query_deduplication: true # Enable query deduplication for all subgraphs.
+-     subgraphs:
+-       products:
+-         query_deduplication: false # Disable query deduplication for products.
++ traffic_shaping:
++   variables_deduplication: true # Enable the variables deduplication optimization
++   all:
++     query_deduplication: true # Enable query deduplication for all subgraphs.
++   subgraphs:
++     products:
++       query_deduplication: false # Disable query deduplication for products.
+```
+
+### Rhai plugin `request.sub_headers` renamed to `request.subgraph.headers` [PR #1261](https://github.com/apollographql/router/pull/1261)
+
+Rhai scripts previously supported the `request.sub_headers` attribute so that subgraph request headers could be
+accessed. This is now replaced with an extended interface for subgraph requests:
+
+```
+request.subgraph.headers
+request.subgraph.body.query
+request.subgraph.body.operation_name
+request.subgraph.body.variables
+request.subgraph.body.extensions
+request.subgraph.uri.host
+request.subgraph.uri.path
+```
+
+By [@garypen](https://github.com/garypen) in https://github.com/apollographql/router/pull/1261
+
+## 🚀 Features
+
+### Add support of compression [PR #1229](https://github.com/apollographql/router/pull/1229)
+Add support of request and response compression for the router and all subgraphs. The router is now able to handle `Content-Encoding` and `Accept-Encoding` headers properly. Supported algorithms are `gzip`, `br`, `deflate`.
+You can also enable compression on subgraphs requests and responses by updating the `traffic_shaping` configuration:
+
+```yaml
+traffic_shaping:
+  all:
+    compression: br # Enable brotli compression for all subgraphs
+  subgraphs:
+    products:
+      compression: gzip # Enable gzip compression only for subgraph products
+```
+
+By [@bnjjj](https://github.com/bnjjj) in https://github.com/apollographql/router/pull/1229
+
+### Add support of multiple uplink URLs [PR #1210](https://github.com/apollographql/router/pull/1210)
+Add support of multiple uplink URLs with a comma-separated list in `APOLLO_UPLINK_ENDPOINTS` and for `--apollo-uplink-endpoints`
+
+Example:
+```bash
+export APOLLO_UPLINK_ENDPOINTS="https://aws.uplink.api.apollographql.com/, https://uplink.api.apollographql.com/"
+```
+
+By [@bnjjj](https://github.com/bnjjj) in https://github.com/apollographql/router/pull/1210
+
+### Add support for adding extra enviromental variables and volumes to helm chart [PR #1245](https://github.com/apollographql/router/pull/1245)
+The following example will allow you to mount your supergraph.yaml into the helm deployment using a configmap with a key of supergraph.yaml. Using [Kustomize](https://kustomize.io/) to generate your configmap from your supergraph.yaml is suggested.
```

Comment:
I don't understand this:

> with a key of supergraph.yaml

Where is `supergraph.yaml` the key?

Reply:
> I don't understand this:
> 
> > with a key of supergraph.yaml
> 
> Where is `supergraph.yaml` the key?

It's a reference to the content of a notional configMap called `some-configmap`, which contains the contents of your config, which will be mounted as a file in the container at `/etc/apollo/supergraph.yaml`.

Reply:
That is exactly my perception of the behavior, but I still don't understand the "key" terminology here.  Could you help me with that?

Reply:
Ah, I understand now with your edit and the example.  Do you think there's a treatment here we could do to make that more clear?

Reply:
Maybe something like:

+++START+++

The following example will allow you to mount your supergraph.yaml into the helm deployment using a configmap which contains an entry identifying the contents of supergraph.yaml. Using [Kustomize](https://kustomize.io/) to generate your configmap from your supergraph.yaml is suggested.

Example:
```yaml
extraEnvVars:
  - name: APOLLO_ROUTER_SUPERGRAPH_PATH
    value: /etc/apollo/supergraph.yaml
    # sets router log level to debug
  - name: APOLLO_ROUTER_LOG
    value: debug
extraEnvVarsCM: ''
extraEnvVarsSecret: ''

extraVolumes:
  - name: supergraph-volume
    configMap:
      name: some-configmap 
extraVolumeMounts: 
  - name: supergraph-volume
    mountPath: /etc/apollo
```
The some-configmap yaml would look something like:
```yaml
supergraph.yaml:
    server:
        listen: 0.0.0.0:80
```
+++END+++ #

Reply:
I've made some edits to my suggestion above that might work for me.  I think that there's a larger conversation here that's out-of-scope for the `CHANGELOG.md`, perhaps. 😄 

Consider me satisfied either way!

Reply:
The phrasing you offered also helps quite well!

Reply:
Tweaked, how does is look now?